### PR TITLE
version 0.5.0

### PR DESCRIPTION
**Pull request by gcanti**
Compatible with

"fp-ts": "^2.0.0"
"io-ts": "^2.0.0"
"monocle-ts": "^2.0.0"

- **Breaking Change**
  - upgrade to `fp-ts@2`, `monocle-ts@2` and `io-ts@2.0.0` (@gcanti)
  - move `fp-ts@2`, `monocle-ts@2` and `io-ts@2.0.0` to `peerDependencies` (@gcanti)
  - remove `boolean` folder (@gcanti)
    - move `BooleanFromString` to top level
  - remove `date` folder (@gcanti)
    - move `date` to top level
    - move `DateFromISOString` to top level
    - move `DateFromNumber` to top level
    - move `DateFromUnixTime` to top level
  - remove `fp-ts` folder (@gcanti)
    - move `createEitherFromJSON` to top level and rename to `either`
    - move `createNonEmptyArrayFromArray` to top level and rename to `nonEmptyArray`
    - move `createOptionFromJSON` to top level and rename to `option`
    - remove `createStrMapFromDictionary`
    - move `createOptionFromNullable` to top level and rename to `optionFromNullable`
    - move `createSetFromArray` to top level and rename to `setFromArray`
  - remove `JSON` folder (@gcanti)
  - remove `monocle-ts` (@gcanti)
    - move `lensesFromInterface` to top level and renamed to `getLenses` (@gcanti)
    - remove `TypePrismIso` module (@gcanti)
  - remove `newtype-ts` folder (@gcanti)
  - remove `number` folder (@gcanti)
    - move `NumberFromString` to top level
    - move `IntegerFromString` to top level and rename to `IntFromString`
  - rename `eitherFromJSON` to `either` (@gcanti)
  - rename `optionFromJSON` to `option` (@gcanti)
  - uncurry `fromNullable` (@gcanti)
  - uncurry `fallback` and rename to `withFallback` (@gcanti)

### CODE REVIEW

File: src/either.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export function either<L extends t.Mixed, R extends t.Mixed>(
+  leftCodec: L,
+  rightCodec: R,
+  name: string = `Either<${leftCodec.name}, ${rightCodec.name}>`
+): EitherC<L, R> {
+  return t.union(
+    [
+      t.strict(
+        {
+          _tag: t.literal('Left'),
```
Comment (DenisFrezzato) [inline src/either.ts]: This literal (and `'Right'` one) can be created once, outside this function. What do you think?

File: test/BooleanFromString.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { BooleanFromString } from '../src/BooleanFromString'
+import { assertSuccess, assertFailure } from './helpers'
+
+describe('BooleanFromString', () => {
+  it('decode', () => {
+    const T = BooleanFromString
+    assertSuccess(T.decode('true'), true)
+    assertSuccess(T.decode('false'), false)
```
Comment (DenisFrezzato) [inline test/BooleanFromString.ts:8]: You should split this test in two different tests, one for `'true'` and one for `'false'`.

**Review by DenisFrezzato — APPROVED** [review]
I noticed you grouped the tests in `name`, `is`, `decode` and `encode` test cases. Actually, test cases should test only one specific behaviour, so they should be splitted.
But overall looks fine.
I didn't pay attention to the documentation (it's generated, isn't it?).

**Comment by gcanti** [comment]
@DenisFrezzato thanks for reviewing (the documention is generated)

**Pull request merged**